src/ecommerce.interfaces.ts

```diff
@@ -96,28 +96,69 @@ export interface SDKECommerceAPI extends IECommerceShared {
         customFlags?: SDKEventCustomFlags,
         eventOptions?: SDKEventOptions
     ): void;
+    setCurrencyCode(code: string): void;
+
+    /*
+     * @deprecated
+     */
     logPurchase(
         transactionAttributes: TransactionAttributes,
         product: SDKProduct | SDKProduct[],
         clearCart?: boolean,
         attrs?: SDKEventAttrs,
         customFlags?: SDKEventCustomFlags
     ): void;
+    /*
+     * @deprecated
+     */
     logRefund(
         transactionAttributes: TransactionAttributes,
         product: SDKProduct | SDKProduct[],
         clearCart?: boolean,
         attrs?: SDKEventAttrs,
         customFlags?: SDKEventCustomFlags
     ): void;
-    setCurrencyCode(code: string): void;
+}
+
+interface ExtractedActionAttributes {
+    "Affiliation"?: string;
+    "Coupon Code"?: string;
+    "Total Amount"?: number;
+    "Shipping Amount"?: string;
```

Comment:
this gets sanitized and is a number - https://github.com/mParticle/mparticle-web-sdk/blob/master/src/ecommerce.js#L28-L31